prov/psm2/src/psmx2_tagged.c

```diff
@@ -996,7 +1065,10 @@ ssize_t psmx2_tagged_sendv_generic(struct fid_ep *ep,
 	}
 
 	av = ep_priv->av;
-	if (av && av->type == FI_AV_TABLE) {
+	if (av && PSMX2_SEP_ADDR_TEST(dest_addr)) {
+		psm2_epaddr = psmx2_av_translate_sep(av, ep_priv->trx_ctxt, dest_addr);
+		vlane = 0;
+	} else  if (av && av->type == FI_AV_TABLE) {
```

Comment:
I think it would be better handled via a separate PR. The consolidation will be applied not only to the newly added code, but also the existing code in the other conditional branches.